identity/extension_verify.go

```diff
@@ -26,42 +25,54 @@ func (r *SchemaExtensionVerification) Run(ctx jsonschema.ValidationContext, s sc
 	defer r.l.Unlock()
 
 	switch s.Verification.Via {
-	case "email":
+	case AddressTypeEmail:
 		if !jsonschema.Formats["email"](value) {
 			return ctx.Error("format", "%q is not valid %q", value, "email")
 		}
 
 		address := NewVerifiableEmailAddress(fmt.Sprintf("%s", value), r.i.ID)
 
-		if has := r.has(r.i.VerifiableAddresses, address); has != nil {
-			if r.has(r.v, address) == nil {
-				r.v = append(r.v, *has)
-			}
-			return nil
-		}
+		r.appendAddress(address)
 
-		if has := r.has(r.v, address); has == nil {
-			r.v = append(r.v, *address)
-		}
+		return nil
+
+	case AddressTypePhone:
+		address := NewVerifiablePhoneAddress(fmt.Sprintf("%s", value), r.i.ID)
```

Comment:
Is there some canonical verification type (e.g. RFC) for this?

Reply:
I think that we can use some implementation of libphonenumber for validation purposes, for example [one](https://github.com/nyaruka/phonenumbers) that was mentioned in issue. 
But we need additional work to implement validation in proper way, so decided to not include it in this PR.
I'm thinking that we need to extend JSON schema [library](https://github.com/ory/jsonschema) with "phone" type and use the phonenumber validation in there.
We can create a separate issue and implement the validation this way or any other in separate PR, what do you think?

Reply:
Yes, that sounds good to me! The problem is usually that we can get hung up on this. We have problem with SMTP already where using an invalid email address (e.g. `asdf`) can lead to the SMTP server rejecting the request, which in turn causes these addresses to be retried (currently forever). So I think we could alleviate this problem a bit with validation.

Reply:
Got it. Okay, will submit PR with phone number support to ory/jsonschema then.

Reply:
Created a PR, please review: https://github.com/ory/jsonschema/pull/7 Once it's merged, we can merge this one too, I think